multimodal/src/autogluon/multimodal/data/process_document.py

```diff
@@ -0,0 +1,331 @@
+
+        assert len(words) == len(boxes)
+
+        results["words"] = words
+        results[BBOX] = boxes
+        return results
+
+    def process_one_sample(
+        self,
+        document_features: Dict[str, Union[NDArray[(Any,), Any], list]],
+        feature_modalities: Dict[str, Union[NDArray[(Any,), Any], list]],
+        is_training: bool,
+        image_mode: Optional[str] = "RGB",
+    ):
+        """
+        Read documents, process them, and stack them. One sample has one document image.
+
+        Parameters
+        ----------
+        document_features
+            One sample has one document image column in a pd.DataFrame.
+        feature_modalities
+            What modality each column belongs to.
+        is_training
+            Whether to process document images in the training mode.
+        image_mode
+            A string which defines the type and depth of a pixel in the image.
+            For example, RGB, RGBA, CMYK, and etc.
+
+        Returns
+        -------
+        A dictionary containing one sample's document and its features.
+        """
+        ret = {}
+        for per_col_name, per_col_image_features in document_features.items():
+            # Open the document image.
+            try:
+                doc_image = PIL.Image.open(per_col_image_features[0])
+            except Exception as e:
+                raise e
+
+            # The OCR process is time-consuming, so apply OCR on each image only once.
+            if per_col_image_features[0] not in self.documents:
+                ocr_res = self.apply_ocr(doc_image)
+                # store the ocr results.
+                self.documents.update({per_col_image_features[0]: ocr_res})
+            else:
+                # reuse the ocr results.
+                ocr_res = self.documents[per_col_image_features[0]]
+
+            words = ocr_res["words"]
+            doc_image = doc_image.convert(image_mode)
+
+            if is_training:
+                doc_image = self.train_processor(doc_image)
+            else:
+                doc_image = self.val_processor(doc_image)
+
+            normalized_word_boxes = ocr_res[BBOX]
+
+            if self.is_text_only_flag:
+                # Padding of token_boxes up the bounding boxes to the sequence length.
+                token_boxes = []
+                all_tokens = []
+                for word, box in zip(words, normalized_word_boxes):
+                    # Tokenize bounding box seperately.
+                    word_tokens = self.tokenizer.tokenize(word)
+                    all_tokens.append(word_tokens)
+                    token_boxes.extend([box] * len(word_tokens))
+
+                encoding = self.tokenizer(
+                    " ".join(words), padding="max_length", truncation=True, return_token_type_ids=True
+                )
+
+                # Truncation of token_boxes
+                pad_token_box = [0, 0, 0, 0]
+                special_tokens_count = 2
```

Comment:
Maybe better to comment on what are the two special tokens.

Reply:
Will add it, thanks!